### DISCUSSION TITLE
Integrate tree-sitter with IDE

### DISCUSSION THREAD

**Discussion by MrChaker** [tree-sitter/tree-sitter#2903] [Q&A]
I'm new to tree-sitter , I've wrote a parser for some markup language for a project , I'm confused now how do I integrate it with vs code . I guess I'll need to publish an extension , But I didn't find how I can do that .

**Comment by sogaiu** [comment]
Last I checked (which admittedly was a while back), VSCode wasn't particularly oriented toward using tree-sitter.  Various folks have managed though.

Some things that might be of interest:

* https://github.com/fallenwood/syntax-highlighter
* https://github.com/cursorless-dev/vscode-parse-tree
* https://github.com/georgewfraser/vscode-tree-sitter ([deprecated effort](https://github.com/georgewfraser/vscode-tree-sitter/issues/44))

May be there are other relevant things that have come about in recent times too (^^;

I think in a number of cases people have gone for integrating via web-tree-sitter instead of node-tree-sitter for at least two reasons:

1. don't have to build the extension for multiple platforms
2. node-tree-sitter had a long period of not really being looked after - this may not be the case these days (verification probably a good idea)

A notable drawback with this approach is that wasm-based things are typically slower than node-based.  At least that's what I recall.  Still, the evidence suggests that it's fast enough in multiple cases.
